lib/commandhandlers.go

```diff
@@ -97,6 +98,45 @@ func loadClientCommands() {
 		minParams:    1,
 		handler: func(listener *Listener, msg ircmsg.IrcMessage) {
 			// We're starting CAP negotiations so don't complete regisration until then
+			listener.regLocks.Set("cap", false)
+
+			command := strings.ToUpper(getParam(&msg, 0))
+			if command == "LS" {
+				capList := Capabilities.AsString()
+				listener.Send(nil, "", "CAP", "*", "LS", capList)
+
+			} else if command == "REQ" {
+				caps := strings.Split(getParam(&msg, 1), " ")
+				acked := []string{}
+				for _, cap := range caps {
+					capVal, isAvailable := Capabilities.Supported[cap]
+					if isAvailable {
+						listener.Caps[cap] = capVal
+						acked = append(acked, cap)
+					}
+				}
+
+				for _, cap := range acked {
+					Capabilities.InitCapOnListener(listener, cap)
+				}
+
+				listener.Send(nil, "", "CAP", "*", "ACK", strings.Join(acked, " "))
+
+			} else if command == "ENABLED" {
+				// Not in the spec, but just a handy command to debug caps in the client
+				line := ""
+				for cap, val := range listener.Caps {
+					line += cap
+					if val != "" {
+						line += "=" + val
+					}
+					line += " "
+				}
+				listener.SendLine(fmt.Sprintf(":%s NOTICE %s :%s", listener.Manager.Source, listener.ClientNick, line))
+
+			} else if command == "END" {
+				listener.regLocks.Set("cap", true)
```

Comment:
Hmm, this is cool but I feel like we should have a bigger discussion around how to authenticate to goshubnc when connecting into it (possible SASL client -> goshubnc and etc), I'll make a new issue to discuss that.